src/graphic/Text.ts

```diff
@@ -665,6 +665,16 @@ class ZRText extends Displayable<TextProps> implements GroupLike {
             let rightIndex = tokenCount - 1;
             let token;
 
+            if (verticalAlign === 'middle') {
```

Comment:
https://github.com/ecomfe/zrender/blob/7ca4a904a29ac50fa126a33b86909782a208dea6/src/graphic/Text.ts#L639 should have done this calculation.

Reply:
No, `adjustTextY` solved the location of background by `verticalAlign = 'top'`, not `middle`.

https://github.com/ecomfe/zrender/blob/7ca4a904a29ac50fa126a33b86909782a208dea6/src/graphic/Text.ts#L727-L729

we need adjust this by `middle` for tokens.

Reply:
test case.

```
const option = {
    series: [{
        type: 'treemap',
        name: '中国',
        leafDepth: 1,
        label: {
            backgroundColor: 'pink',
            width: 200,
            height: 100,
            align: 'right',
            verticalAlign: 'bottom',
            formatter: '{b|{a}}{c}{a|{b}}',
            rich: {
                a: {
                    color: 'red',
                    verticalAlign: 'bottom',
                    backgroundColor: 'green',
                },
                b: {
                    color: 'red',
                    verticalAlign: 'top',
                    backgroundColor: 'green',
                }
            }
        },
        data: [
            {
                name: '河南省',
                value: 20,
                itemStyle: {
                    color: 'magenta'
                },
                children: [{
                    name: '郑州市',
                    value: 20,
                    children: [{
                        name: 'xxx县',
                        value: 20
                    }]
                }]
            }
        ]
    }]
};
```